_stbt/imgproc_cache.py

```diff
@@ -0,0 +1,211 @@
+import functools
+import inspect
+import json
+import os
+import sys
+from contextlib import contextmanager
+
+import numpy
+
+from _stbt.gst_utils import Gst, numpy_from_sample
+from _stbt.logging import ImageLogger
+from _stbt.utils import mkdir_p
+
+_cache = None
+
+
+@contextmanager
+def cache(filename=None):
+    if os.environ.get('STBT_DISABLE_CACHING'):
+        yield
+        return
+
+    global _cache
+
+    sys.path.append(os.path.dirname(__file__))
+    import _stbt.lmdb as lmdb
+    del sys.path[-1]
+
+    if filename is None:
+        cache_home = os.environ.get('XDG_CACHE_HOME') \
+            or '%s/.cache' % os.environ['HOME']
+        mkdir_p(cache_home + "/stbt")
+        filename = cache_home + "/stbt/cache.lmdb"
+    with lmdb.open(filename) as db:  # pylint: disable=no-member
+        assert _cache is None
```

Comment:
I'm happy to add a message.  "concurrently" isn't quite the right term though as this will happen if you do:

    with cache():
        with cache():
            ...